src/scala/microsoft-spark-2-3/src/test/scala/org/apache/spark/api/dotnet/Extensions.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed to the .NET Foundation under one or more agreements.
+ * The .NET Foundation licenses this file to you under the MIT license.
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.apache.spark.api.dotnet
+
+import java.io.DataInputStream
+
+object Extensions {
+  implicit class DataInputStreamExt(stream: DataInputStream) {
+    def readN(n: Int): Array[Byte] = {
```

Comment:
nit: `readNBytes`. Since this is only used inside `SerDeTests`, could you just define a helper function inside the test suite instead? (When I saw `in.readN`, I thought it was a public API)

Reply:
Done as suggested for 2.3.
In 2.4 and 3.0 there are multiple usages of `readN` -> `readNBytes`:
- SerDeTest
- DotnetBackendHandlerTest
So for these versions I've took the liberty of leaving this extension method as is, but with making it visible only for `dotnet` package.
Anyway I'm happy to hear you thoughts about that.

Reply:
Ah, in that case, let's leave it as an extension method even for 2.3 so that it's easier to maintain.

Reply:
I've added the extension method back for 2.3.